Added design for long mirror with join traffic (#5000)
telstra/open-kilda: docs/design/flow-traffic-mirroring/flow-traffic-mirroring.md

  - Need to add information about the state of the mirror paths in the flow payload.
  - It is necessary to add information about built mirror paths to API `GET /flows/{flow_id}/path`

## Joining traffic

> It is possible to make mirror bidirectional. With special flag `join_traffic_from_sink=true` 

Reviewer: ```suggestion
It is possible to make a mirror bidirectional. With a special flag `join_traffic_from_sink=true` 
```

> user can join traffic from sink to flow traffic in opposite direction.

Reviewer: ```suggestion
users can join traffic from sink to flow traffic in the opposite direction.
```

The group will have 2 or more buckets: one will represent output to the flow or ISL port, 
the rest will represent mirror actions set (i.e. routing to the mirror paths).

## Need to decide

> 1. What status flow should have if mirror path is down

Reviewer: ```suggestion
1. Which status a flow should have when a mirror path is in the `down` state
```

> 2. What path computation strategy mirror path should use (guess flow strategy)

Reviewer: ```suggestion
2. Which path computation strategy a mirror path should use (guess flow strategy)
```

> 3. Should we use ingress meter for egress sink traffic 

Reviewer: ```suggestion
3. Should we use an ingress meter for egress sink traffic? 
```

## FSM diagrams

### FlowMirrorPointCreateFsm
![FlowMirrorPointCreateFsm](./flow-create-mirror-point-fsm.png "FlowMirrorPointCreateFsm")

### FlowMirrorPointDeleteFsm
![FlowMirrorPointDeleteFsm](./flow-delete-mirror-point-fsm.png "FlowMirrorPointDeleteFsm")

## Traffic examples

> NOTE: in all examples `join_traffic_from_sink` is `true`. With `join_traffic_from_sink=false`

Reviewer: ```suggestion
NOTE: in all examples `join_traffic_from_sink` is `true`. With `join_traffic_from_sink=false`,
```

> there would no traffic from mirror (green arrow).

Reviewer: ```suggestion
there would be no traffic from the mirror (green arrows).
```
or `there is no`
